### PR TITLE
Update to the latest payments code

### PR DESCRIPTION
**Pull request by bluk**
- Updates to the latest payments API spec
- Adds the `ApplicationContext` object to Payments

### CODE REVIEW

File: Source/v1/Payments/Currency.cs [hunk 1]
```diff
@@ -22,14 +22,15 @@ public class Currency
         public Currency() {}
 
         /// <summary>
+        /// REQUIRED
         /// The [three-character ISO-4217 currency code](/docs/integration/direct/rest/currency-codes/). PayPal does not support all currencies.
         /// </summary>
         [DataMember(Name="currency", EmitDefaultValue = false)]
-        public string CurrencyCode;
```
Comment (demerino) [inline Source/v1/Payments/Currency.cs]: This is causing an issue in CI. 

> v1/Payments/Currency.cs(29,23): error CS0542: 'Currency': member names cannot be the same as their enclosing type

File: Source/v1/Payments/DetailedRefund.cs [hunk 2]
```diff
@@ -57,6 +57,12 @@ public class DetailedRefund
         [DataMember(Name="id", EmitDefaultValue = false)]
         public string Id;
 
+        /// <summary>
+        /// The invoice number to track this payment.
+        /// </summary>
+        [DataMember(Name="invoice_number", EmitDefaultValue = false)]
+        public string InvoiceNumber;
```
Comment (demerino) [inline Source/v1/Payments/DetailedRefund.cs]: CI issue here: 
> v1/Payments/DetailedRefund.cs(70,23): error CS0102: The type 'DetailedRefund' already contains a definition for 'InvoiceNumber'

File: Source/v1/Payments/PaymentHoldReason.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        /// <summary>
+	    /// Required default constructor
+		/// </summary>
+        public PaymentHoldReason() {}
+
+        /// <summary>
+        /// The reason that PayPal holds the recipient fund. Set only if the payment hold status is `HELD`.
+        /// </summary>
+        [DataMember(Name="payment_hold_reason", EmitDefaultValue = false)]
+        public string PaymentHoldReason;
```
Comment (demerino) [inline Source/v1/Payments/PaymentHoldReason.cs]: Enclosing type issue.
> v1/Payments/PaymentHoldReason.cs(28,23): error CS0542: 'PaymentHoldReason': member names cannot be the same as their enclosing type

**Comment by HoneyShyam** [comment]
Hope this commit have a custom for subscription payments

**Comment by bluk** [comment]
@HoneyShyam Is there a specific field you are looking for?

**Comment by bluk** [comment]
The Travis-CI failures are due to current PayPal sandbox issues.

**Pull request merged**